modules/afsmtp/afsmtp.c

```diff
@@ -382,24 +393,28 @@ __build_message(AFSMTPDriver *self, LogMessage *msg, smtp_session_t session)
 
   message = smtp_add_message(session);
 
-  smtp_set_reverse_path(message, self->mail_from->address);
+  log_template_format(self->mail_from->addr_tmpl, msg, &self->template_options, LTZ_SEND,
+                      self->super.seq_num, NULL, self->str);
+  smtp_set_reverse_path(message, afsmtp_wash_string(self->str->str));
```

Comment:
Hmm probably reverse_path should take care about  template formatting, and not open coding it numerous times.

How does memory ordership go? What frees the return value of wash_string?